doc/user_guide/options.rst

```diff
@@ -82,6 +84,15 @@ Following options are exposed:
 
 .. option:: --inlinevar-naming-style=<style>
 
+Predefined Naming Patterns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Pylint provides predefined naming patterns for some names. These patterns are often
+based on a Naming Style but there is no option to choose one of the styles mentioned above.
+The pattern can be overwritten with the options discussed below.
+
+The following type of names are checked with a predefined pattern:
+
+- TypeVars. Accepted names include: `T`, `TypeVar`, `_CallableT`, `_T_co`.
```

Comment:
I wonder if a table would make sense here? With `Name type`, `Good names`, and `Bad names`.

I would prefer not to suggest `TypeVar`. True it's valid, but it shouldn't be used IMO. Let's use something like `MyClassT` to highlight PascalCase. For bad names, we should definitely mention something like `T_Class`, and `MyClassType`.

Some other names from the tests: `AnyStr`, `DeviceTypeT`, `DeviceType`, `CALLABLE_T`.